core/src/com/smeanox/games/sg002/world/GameWorld.java

```diff
@@ -148,26 +151,35 @@ public boolean isPlayerStillAlive(Player player){
 		return false;
 	}
 
+
 	/**
-	 * Checks if the given GameObject was used
+	 * removes the GameObject from the map
 	 * @param x coordinates
 	 * @param y coordinates
-	 * @return true if it was used
 	 */
-	public boolean wasUsed(int x, int y){
-		if(getWorldMap(x, y) == null){
-			return false;
-		}
-		return usedGameObjects.contains(getWorldMap(x, y));
+	public void removeGameObject(int x, int y){
+		worldMap[y][x] = null;
 	}
 
 	/**
-	 * removes the GameObject from the map
+	 * Checks whether a gameObject has any remaining actions
 	 * @param x coordinates
 	 * @param y coordinates
+	 * @return
 	 */
-	public void removeGameObject(int x, int y){
-		worldMap[y][x] = null;
+	public boolean wasUsed(int y, int x){
+		try {
+			/*if (x < 0 || y < 0 || x >= mapSizeX || y >= mapSizeY) {
+				return true;//what happened here?
+			}
+			if (worldMap[x][y] == null) return true; //this shouldn't happen either*/
```

Comment:
True, this shouldn't happen. But `catch (Exception ex)` is not really nice. I think an assertion would fit better.